pkgs/by-name/pr/proton-ge-custom/package.nix

```
@@ -0,0 +1,42 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, writeScript
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "proton-ge-custom";
+  version = "GE-Proton9-1";
+
+  src = fetchurl {
+    url = "https://github.com/GloriousEggroll/proton-ge-custom/releases/download/${finalAttrs.version}/${finalAttrs.version}.tar.gz";
+    hash = "sha256-wCIffeayOy3kEwmIKB7e+NrliuSpKXoVYC334fxVB3U=";
+  };
+
+  buildCommand = ''
+    runHook preBuild
+    mkdir -p $out/{bin,opt}
+    tar -C $out/opt --strip=1 -x -f $src
+    ln -s $out/opt/toolmanifest.vdf $out/bin/toolmanifest.vdf
+    install -Dm644 $out/opt/compatibilitytool.vdf $out/bin/compatibilitytool.vdf
+    substituteInPlace $out/bin/compatibilitytool.vdf \
+      --replace-fail "\"install_path\" \".\"" "\"install_path\" \"$out/opt\""
```

Nit: A bit clearer to read:

```suggestion
    substituteInPlace $out/bin/compatibilitytool.vdf \
      --replace-fail '"install_path" "."' '"install_path" "${placeholder "out"}/opt"'
```